doc/lua_api.md

```diff
@@ -4191,43 +4200,79 @@ minetest.register_on_joinplayer(function(player)
+            "You have been playing for @1 minutes.",
+            minutes, tostring(minutes))
+    end,
+})
 ```
 
 When someone called "CoolGuy" joins the game with an old client or a client
 that does not have localization enabled, they will see `Hello CoolGuy, how are
-you today?`
+you today?`. If they use the `/playtime` command, they will see `You have been
+playing for 1 minute` or (for example) `You have been playing for 4 minutes.`
 
-However, if we have for instance a translation file named `hello.de.tr`
+However, if we have for instance a translation file named `hello.de.po`
 containing the following:
 
-    # textdomain: hello
-    Hello @1, how are you today?=Hallo @1, wie geht es dir heute?
+```po
+msgid ""
+msgstr ""
+"Plural-Forms: nplurals=2; plural=(n != 1);\n"
+
+msgid "Hello @1, how are you today?"
+msgstr "Hallo @1, wie geht es dir heute?"
+
+msgid "You have been playing for @1 minute."
+msgid_plural "You have been playing for @1 minutes."
+msgstr[0] "Du spielst seit @1 Minute."
+msgstr[1] "Du spielst seit @1 Minuten."
+```
 
 and CoolGuy has set a German locale, they will see `Hallo CoolGuy, wie geht es
-dir heute?`
+dir heute?` when they join, and the `/playtime` command will show them `Du
+spielst seit 1 Minute.` or (for example) `Du spielst seit 4 Minuten.`
 
-* `minetest.translate_n(textdomain, str, str_plural, n, ...)` translates the
-  string `str` with the given `textdomain` for disambiguaion. The value of
-  `n`, which must be a nonnegative integer, is used to decide whether to use
-  the singular or the plural version of the string. Depending on the locale of
-  the client, the choice between singular and plural might be more complicated,
-  but the choice will be done automatically using the value of `n`.
+Creating and updating translation files
+---------------------------------------
 
-  You can read https://www.gnu.org/software/gettext/manual/html_node/Plural-forms.html
-  for more details on the differences of plurals between languages.
+As an alternative to writing translation files by hand (as shown in the above
+example), it is also possible to generate translation files based on the source
+code.
 
-  Also note that plurals are only handled in .po or .mo files, and not in .tr files.
+It is recommended to first generate a translation template. The translation
+template includes translatable strings that translators can directly work on.
+After creating the `locale` directory, a translation template for the above
+example using the following command:
 
-  A typical use of it looks like:
-  ```lua
-  local S, NS = minetest.get_translator()
-  NS("@1 file", "@1 files", n, tostring(n))
-  ```
-  Here, "1 file" will be displayed if `n` is 1, but `[n] files` will be displayed
-  otherwise. On clients with a different locale, the translation file will provide
-  different translation strings depending on the value of `n`.
+```sh
+xgettext -L lua -kS -kNS:1,2 -d hello -o locale/hello.pot *.lua
```

Comment:
I think this will fail to generate `msgctxt` entries; but I'm not sure how to generate them with `xgettext`. Maybe we should at least add `-k minetest.translate:1c,2 -k minetest.translate_n:1c,2,3` as additional arguments to `xgettext` so that those are correctly handled?

Reply:
The previous PR implemented the feature that allows inferring the textdomain from the filename if the `msgctxt` entry is missing (https://github.com/minetest/minetest/pull/13687#discussion_r1518407844) so it is IMO fine to not generate `msgctxt` for `S` and `NS`.

That said, it does makes sense to add `msgctxt` entries for `minetest.translate` and `minetest.translate_n`.

Reply:
I updated the documentation in https://github.com/minetest/minetest/pull/14726/commits/6b269039194e25f929130e22e3609bbce35cd537